test/array.test.ts

```diff
@@ -40,4 +40,12 @@ describe('array', () => {
 
     expectRejectValues(runtype, [undefined, null, ['asd'], [undefined, 1], '1'])
   })
+
+  it('ignores keys already ignored by sloppyRecord', () => {
```

Comment:
Looks like this test is better moved to sloppyRecord.test.ts?

But I see that an impure test is missing for arrays, could you please add one using the string runtype (because it is the most basic impure runtype) and use the `expectValuesImpure` helper?